File: app/src/main/java/com/checkin/app/checkin/Shop/Private/Invoice/ShopInvoiceDetailActivity.java
package com.checkin.app.checkin.Shop.Private.Invoice;

import android.os.Bundle;
import android.widget.TextView;

import com.checkin.app.checkin.misc.models.BriefModel;
import com.checkin.app.checkin.R;
import com.google.android.material.appbar.AppBarLayout;
import com.google.android.material.tabs.TabLayout;

import java.util.Locale;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.lifecycle.ViewModelProviders;
import androidx.viewpager.widget.ViewPager;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class ShopInvoiceDetailActivity extends AppCompatActivity {
    public static final String KEY_SESSION_DATA = "shop.invoice_session";

    @BindView(R.id.tv_invoice_session_id)
    TextView tvSessionId;
    @BindView(R.id.tv_invoice_session_date)
    TextView tvDate;
    @BindView(R.id.tv_invoice_session_item_count)
    TextView tvItemCount;
    @BindView(R.id.tv_invoice_session_member_count)
    TextView tvMemberCount;
    @BindView(R.id.tv_invoice_session_waiter)
    TextView tvWaiter;
    @BindView(R.id.tv_invoice_session_table)
    TextView tvTable;
    @BindView(R.id.appbar_shop_invoice)
    AppBarLayout appBarLayout;
    @BindView(R.id.tabs_shop_invoice)
    TabLayout tabLayout;
    @BindView(R.id.pager_shop_invoice)
    ViewPager pagerInvoice;

    private ShopSessionViewModel mViewModel;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_shop_invoice_detail);
        ButterKnife.bind(this);

        RestaurantSessionModel data = (RestaurantSessionModel) getIntent().getSerializableExtra(KEY_SESSION_DATA);

        if (data != null) {
            setupAppbarDetails(data);
            setupViewPager();
            setupData(data);
        }
    }

    private void setupData(RestaurantSessionModel data) {
        mViewModel = ViewModelProviders.of(this).get(ShopSessionViewModel.class);
        mViewModel.setSessionPk(data.getPk());
    }

    private void setupAppbarDetails(RestaurantSessionModel data) {
        BriefModel host = data.getHost();
        tvWaiter.setText(String.format(Locale.ENGLISH, "Waiter : %s", host != null ? host.getDisplayName() : getResources().getString(R.string.waiter_unassigned)));

        tvSessionId.setText(data.getHashId());
        tvDate.setText(data.getFormattedDate());
        tvMemberCount.setText(String.valueOf(data.getCountCustomers()));
        tvItemCount.setText(String.format(Locale.ENGLISH, " | %d item(s)", data.getCountOrders()));
        tvTable.setText(data.getTable());
    }

    private void setupViewPager() {
        pagerInvoice.setAdapter(new InvoiceFragmentAdapter(getSupportFragmentManager()));
        tabLayout.setupWithViewPager(pagerInvoice);
    }

    @OnClick(R.id.im_shop_invoice_appbar_back)
    public void onBackClicked() {
        onBackPressed();
    }

    public static class InvoiceFragmentAdapter extends FragmentPagerAdapter {

        public InvoiceFragmentAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public Fragment getItem(int position) {
            switch (position) {
                case 0:
                    return ShopSessionDetailFragment.newInstance();
                case 1:
                    return ShopSessionFeedbackFragment.newInstance();
            }
            return null;
        }

        @Override
        public int getCount() {
            return 2;
        }

        @Nullable
        @Override
        public CharSequence getPageTitle(int position) {
            switch (position) {
                case 0:
                    return "Detail";
                case 1:
                    return "Feedbacks";
            }
            return null;
        }
    }
}

File: app/src/main/java/com/checkin/app/checkin/Shop/Private/Invoice/ShopInvoiceDetailActivity.kt
package com.checkin.app.checkin.Shop.Private.Invoice

import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.activity.viewModels
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import butterknife.BindView
import butterknife.ButterKnife
import butterknife.OnClick
import com.checkin.app.checkin.R
import com.checkin.app.checkin.misc.activities.BaseActivity
import com.checkin.app.checkin.misc.models.BriefModel
import com.google.android.material.tabs.TabLayout
import java.util.*

class ShopInvoiceDetailActivity : BaseActivity() {
    @BindView(R.id.tv_invoice_session_id)
    internal lateinit var tvSessionId: TextView
    @BindView(R.id.tv_invoice_session_date)
    internal lateinit var tvDate: TextView
    @BindView(R.id.tv_invoice_session_item_count)
    internal lateinit var tvItemCount: TextView
    @BindView(R.id.tv_invoice_session_member_count)
    internal lateinit var tvMemberCount: TextView
    @BindView(R.id.tv_invoice_session_waiter)
    internal lateinit var tvWaiter: TextView
    @BindView(R.id.tv_invoice_session_table)
    internal lateinit var tvTable: TextView
    @BindView(R.id.tabs_shop_invoice)
    internal lateinit var tabLayout: TabLayout
    @BindView(R.id.pager_shop_invoice)
    internal lateinit var pagerInvoice: ViewPager

    private val mViewModel: ShopSessionViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_shop_invoice_detail)
        ButterKnife.bind(this)
        val data = intent.getSerializableExtra(KEY_SESSION_DATA) as RestaurantSessionModel
        setupAppbarDetails(data)
        setupViewPager()
        setupData(data)
    }

    private fun setupData(data: RestaurantSessionModel) {
        mViewModel.setSessionPk(data.pk)
    }

    private fun setupAppbarDetails(data: RestaurantSessionModel) {
        val host: BriefModel? = data.host
        if (data.isScheduled) {
            tvWaiter.visibility = View.GONE
        } else {
            tvWaiter.visibility = View.VISIBLE
            tvWaiter.text = String.format(Locale.ENGLISH, "Waiter : %s", if (host != null) host.displayName else resources.getString(R.string.waiter_unassigned))
        }
        tvSessionId.text = data.formatHashId
        tvDate.text = data.formattedDate
        tvMemberCount.text = data.countCustomers.toString()
        tvItemCount.text = String.format(Locale.ENGLISH, " | %d item(s)", data.countOrders)
        tvTable.text = data.tableInfo
    }

    private fun setupViewPager() {
        pagerInvoice.adapter = InvoiceFragmentAdapter(supportFragmentManager)
        tabLayout.setupWithViewPager(pagerInvoice)
    }

    @OnClick(R.id.im_shop_invoice_appbar_back)
    fun onBackClicked() {
        onBackPressed()
    }

    class InvoiceFragmentAdapter(fm: FragmentManager?) : FragmentPagerAdapter(fm!!) {
        override fun getItem(position: Int): Fragment = when (position) {
            0 -> ShopSessionDetailFragment.newInstance()
            else -> ShopSessionFeedbackFragment.newInstance()
        }

        override fun getCount(): Int = 2

        override fun getPageTitle(position: Int): CharSequence? = when (position) {
            0 -> "Detail"
            1 -> "Feedbacks"
            else -> null
        }
    }

    companion object {
        const val KEY_SESSION_DATA = "shop.invoice_session"
    }
}